### PR TITLE
feat: support NuGet package manager

### LINKED ISSUE #20: Add support for nuget
I suggest add a [nuget][nuget] package manager 

Components that might help you:
- [NuGet API Doc][nuget-doc-api]
- [NuGet Client for JS][nuget-cli-npm]

Part of #2 

[nuget]: https://www.nuget.org
[nuget-doc-api]: https://docs.microsoft.com/en-us/nuget/api/overview
[nuget-cli-npm]: https://www.npmjs.com/package/nuget-client

### PR DESCRIPTION
**Pull request by romanrostislavovich**
Added following implimtations for nuget:
- getMeta method
- getDeps method
- getVersions method

Added test and change README file

Closes #20

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,11 @@
     "description": "Registry provides an interface for obtaining information about `packages` from various `package managers`",
     "license": "MIT",
     "keywords": [
-      "linterhub"
+      "linterhub",
```
Comment (xferra) [inline package.json]: Please do not change it unless needed.

Comment (romanrostislavovich) [reply package.json]: I fixed it

File: src/managers/nuget.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import Request from 'request-promise';
+import Meta from '../interface/meta';
+
+export default class extends Manager {
+
+    protected host = 'api.nuget.org/v3';
+
+    protected getVersionData(json : any, version?: string) : any {
+        const actualVersion = version ? version : json.items[0].upper;
+        return json.items[0].items.filter((item : any) => item.catalogEntry.version === actualVersion).shift();
```
Comment (xferra) [inline src/managers/nuget.ts:12]: I suggest simplifying the expression, e.g. split it into several local vars.

File: src/managers/nuget.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    private async requestNuget(name: string, version?: string): Promise<any> {
+        const packageUrl = `https://${this.host}/registration3/${name.toLowerCase()}/index.json`;
```
Comment (xferra) [inline src/managers/nuget.ts:16]: Looks like config param, class or app-wide.
Just FYI: https://github.com/librariesio/package-managers/blob/master/package-managers.json

File: src/managers/nuget.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        };
+    }
+
+    async getDeps(name: string, version?: string): Promise<Dependency[]> {
+        const json = await this.requestNuget(name, version);
+        const data = this.getVersionData(json, version);
+        const dependencies: Dependency[] = [];
+
+        if (data.catalogEntry.dependencyGroups) {
+            data.catalogEntry.dependencyGroups.map((dependencie : any) => {
```
Comment (xferra) [inline src/managers/nuget.ts:44]: `dependency`?

File: src/managers/nuget.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    private async requestNuget(name: string, version?: string): Promise<any> {
+        const packageUrl = `https://${this.host}/registration3/${name.toLowerCase()}/index.json`;
+        return JSON.parse(await Request(packageUrl));
+    }
+
+    constructor(name: string, host: string) {
+        super(name);
+        this.host = host ? host : this.host;
```
Comment (xferra) [inline src/managers/nuget.ts:22]: I suppose TS supports defaults for args?

File: src/managers/nuget.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+    async getMeta(name: string, version?: string): Promise<Meta> {
+        const json = await this.requestNuget(name, version);
+        const data = this.getVersionData(json, version);
+        
+        return {
+            name: data.catalogEntry.title,
+            description: data.catalogEntry.description,
+            url: data.catalogEntry.projectUrl,
+            license: data.catalogEntry.licenseUrl ? data.catalogEntry.licenseUrl : null,
```
Comment (xferra) [inline src/managers/nuget.ts:33]: Is it possible to write smth like `exp || null`? Even if yes, what is the idea? 

`exp != null ? exp : null` Do I understand it properly?

Comment (doroxz-zz) [reply src/managers/nuget.ts:33]: `null` is not `undefined`

**Comment by doroxz-zz** [comment]
Will be added https://github.com/linterhub/registry/pull/33

**Pull request closed without merging**